browser-interface/packages/unity-interface/audioStream.ts

```diff
@@ -1,7 +1,82 @@
 /////////////////////////////////// AUDIO STREAMING ///////////////////////////////////
+import { Entity, IEngine } from '@dcl/ecs/dist-cjs'
+import { AudioEvent as defineAudioEvent, AudioState } from '@dcl/ecs/dist-cjs/components'
+import { SceneWorker } from '../shared/world/SceneWorker'
+import { getSceneWorkerBySceneID, getSceneWorkerBySceneNumber } from '../shared/world/parcelSceneManager'
+
+type AudioEvents = HTMLMediaElementEventMap
+type GlobalProps = {
+  playToken: number
+  entityId: Entity | undefined
+  engine: IEngine | undefined
+}
+
+let globalProps: GlobalProps = {
+  playToken: 0,
+  entityId: undefined,
+  engine: undefined
+}
+
+function setGlobalProps(props: GlobalProps) {
+  globalProps = props
+}
+
+// https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement#events
+const AUDIO_EVENTS: (keyof AudioEvents)[] = [
+  'loadeddata',
+  'error',
+  'seeking',
+  'loadstart',
+  'waiting',
+  'playing',
+  'pause'
+]
 
 const audioStreamSource = new Audio()
-let playToken: number = 0;
+AUDIO_EVENTS.forEach(($) => audioStreamSource.addEventListener($, listen))
+
+function listen(ev: AudioEvents[keyof AudioEvents]) {
```

Comment:
dont feel safe by passing the engine here.
Why dont do the same approach that we have for the comms message ?
Create a mitter and every time there is a change on a listener emit an event with the sceneId number that triggered that event.

Then on the internalEngine we can listen to those changes.
This way we have all the logic of the components INSIDE the internalEngine and no external functions modifiy the engine/components.

Reply:
```ts
// AudioGlobal.ts
const audioMitter = mitt<AudioMitter>()
function listen(ev: AudioEvents[keyof AudioEvents]) {
  audioMitter.emit('change', { sceneId, event })
})

// InternalEngine.ts
import { audioMitter } from '../global-audio.ts'
audioMitter.on('change', ({ sceneId, event}) => { // logic for the components here })
```

Reply:
I agree on having all the `internalEngine` updates on the same place is better (will apply those changes), but something to note about `dont feel safe by passing the engine here` is that all the `internalEngine's` are exposed via [this](https://github.com/decentraland/unity-renderer/blob/70d2e92ef2cad427801a704e3e7401fc00967247/browser-interface/packages/shared/world/parcelSceneManager.ts#L38)

so you can actually access the engine from **anywhere**

<img width="1102" alt="Screenshot 2024-04-09 at 2 58 39 PM" src="https://github.com/decentraland/unity-renderer/assets/2366069/0b5692f1-cdcf-496f-8410-7ef75843f49a">